crates/motionfile/src/motion_interpolator.rs

```diff
@@ -38,6 +39,16 @@ enum State {
         time_since_start: Duration,
     },
     Finished,
+    Aborted {
+        frame_side: Side,
```

Comment:
Is it necessary to remember the `frame_side`? I don't know the details, please explain shortly. Thanks!

Reply:
Yes, this was the origin of the dying robot bug. In case the motion is aborted, the interpolator will be asked for joint angles one last time. In this case the correct joint angles have to be calculated. If the entry condition failed, the initial values of a segment are correct, if the exit condition failed the end values of a segment need to be output. Alternatively `Aborted` could save the `Joints` directly, however this would make the enum variant very large.